x/lockup/ante.go

```diff
@@ -8,30 +12,46 @@ import (
 	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
 	authz "github.com/cosmos/cosmos-sdk/x/authz"
 	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	ibctransfertypes "github.com/cosmos/ibc-go/v3/modules/apps/transfer/types"
+
+	microtxtypes "github.com/althea-net/althea-chain/x/microtx/types"
 
 	"github.com/althea-net/althea-chain/x/lockup/keeper"
 	"github.com/althea-net/althea-chain/x/lockup/types"
 )
 
-// WrappedAnteHandle An AnteDecorator used to wrap any AnteHandler for decorator chaining
+// WrappedAnteHandler An AnteDecorator used to wrap any AnteHandler for decorator chaining
+// This is necessary to use the Cosmos SDK's NewAnteHandler() output with a LockupAnteHandler, as the sdk does not
+// expose a stable list of AnteDecorators before chaining, and every upgrade poses a risk of missing updates
 type WrappedAnteHandler struct {
 	anteHandler sdk.AnteHandler
 }
 
 // AnteHandle calls wad.anteHandler and then the next one in the chain
-// This is necessary to use the Cosmos SDK's NewAnteHandler() output with a LockupAnteHandler
 func (wad WrappedAnteHandler) AnteHandle(
 	ctx sdk.Context,
 	tx sdk.Tx, simulate bool,
 	next sdk.AnteHandler,
-) (newCtx sdk.Context, err error) {
-	modCtx, ok := wad.anteHandler(ctx, tx, simulate)
-	if ok != nil {
+) (sdk.Context, error) {
```

Comment:
This was a significant oversight I made. Using the named return variables with the misnamed `ok` error variable allowed any failure in the wrapped antehandler through instead of returning the error early, since `err` is nil-initialized.

Reply:
I think this is a go vet blindspot since I used the ok variable and then returned the zero intialized err variable.